### PR TITLE
Ports GAGS

### LINKED ISSUE #4533: [$25 BOUNTY] Port GAGS
TG has a new system called Greyscale AutoGenerated Sprites, which enables advanced greyscale usage and generally reduces icon bloat.

There is a *large* number of PRs to port and icon files to edit, hence this bounty. A mostly (but not entirely) complete list of PRs to port can be found as part of https://github.com/BeeStation/BeeStation-Hornet/projects/2#column-7174998

Upon completion and successful testing a single payment of $25 will be sent directly via Paypal, due to the numerous problems with Bountysource. No alternative compensation options will be provided. Whomever completes it should provide their Discord info in the PR description so we can contact you.

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

This PR ports GAGS system from TG

Ported PRs:
https://github.com/tgstation/tgstation/pull/58337
https://github.com/tgstation/tgstation/pull/58487
https://github.com/tgstation/tgstation/pull/58515
https://github.com/tgstation/tgstation/pull/58567
https://github.com/tgstation/tgstation/pull/58794
https://github.com/tgstation/tgstation/pull/58854
https://github.com/tgstation/tgstation/pull/58602
https://github.com/tgstation/tgstation/pull/59114
https://github.com/tgstation/tgstation/pull/59271
https://github.com/tgstation/tgstation/pull/59330
https://github.com/tgstation/tgstation/pull/59524
https://github.com/tgstation/tgstation/pull/59884
https://github.com/tgstation/tgstation/pull/60819
https://github.com/tgstation/tgstation/pull/63957
https://github.com/tgstation/tgstation/pull/58453
https://github.com/tgstation/tgstation/pull/58378

So far ported canister and scredrivers. Rest of the gagified sprites will be ported in future PRs.

## Testing Photographs and Procedure

<details>

obraz

obraz

obraz

obraz

obraz

obraz

obraz

obraz

</details>

Closes #4533

## Changelog
:cl: ninjanomnom, Arkatos1 
add: Added GAGS system, for generating colored icons.
add: GAGS for canisters and screwdrivers.
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
Known issues:
- greyscale menu will runtime when changing from sprite with less colors to a sprite with more colors (for example from canister to proto canister). Issue also present on TG.
- proto canister's third layer won't get recolored, issue present on TG.

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -61,91 +64,106 @@
...
 
-/obj/machinery/portable_atmospherics/canister/oxygen
-	name = "o2 canister"
-	desc = "Oxygen. Necessary for human life."
-	icon_state = "blue"
-	gas_type = GAS_O2
+/obj/machinery/portable_atmospherics/canister/bz
+	name = "\improper BZ canister"
+	desc = "BZ, a powerful hallucinogenic nerve agent."
+	icon_state = "purple"
```
Comment (KubeRoot) [inline code/modules/atmospherics/machinery/portable/canister.dm]: Should probably not have `icon_state` anymore

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 2]
```diff
@@ -61,91 +64,106 @@
...
 
-/obj/machinery/portable_atmospherics/canister/nitryl
-	name = "nitryl canister"
-	desc = "Nitryl gas. Feels great 'til the acid eats your lungs."
-	icon_state = "brown"
-	gas_type = GAS_NITRYL
+/obj/machinery/portable_atmospherics/canister/oxygen
+	name = "o2 canister"
+	desc = "Oxygen. Necessary for human life."
+	icon_state = "blue"
```
Comment (KubeRoot) [inline code/modules/atmospherics/machinery/portable/canister.dm]: Should probably not have `icon_state` anymore

File: tgui/packages/tgui/components/LabeledList.tsx [hunk 3]
```diff
@@ -9,9 +9,9 @@ import { InfernoNode } from 'inferno';
 import { Box, unit } from './Box';
 import { Divider } from './Divider';
 
-type LabeledListProps = {
-  children?: any;
-};
+ type LabeledListProps = {
+   children?: any;
+ };
```
Comment (KubeRoot) [inline tgui/packages/tgui/components/LabeledList.tsx]: Unnecessary preceding whitespace here

File: tgui/packages/tgui/components/LabeledList.tsx [hunk 4]
```diff
@@ -24,17 +24,18 @@ export const LabeledList = (props: LabeledListProps) => {
...
+   label?: string | InfernoNode | BooleanLike;
+   labelColor?: string | BooleanLike;
+   color?: string | BooleanLike;
+   textAlign?: string | BooleanLike;
+   buttons?: InfernoNode,
+   /** @deprecated */
+   content?: any,
+   children?: InfernoNode;
+   verticalAlign?: string;
+ };
```
Comment (KubeRoot) [inline tgui/packages/tgui/components/LabeledList.tsx]: Unnecessary preceding whitespace here too

File: tgui/packages/tgui/components/LabeledList.tsx [hunk 5]
```diff
@@ -85,9 +89,9 @@ const LabeledListItem = (props: LabeledListItemProps) => {
 
 LabeledListItem.defaultHooks = pureComponentHooks;
 
-type LabeledListDividerProps = {
-  size?: number;
-};
+ type LabeledListDividerProps = {
+   size?: number;
+ };
```
Comment (KubeRoot) [inline tgui/packages/tgui/components/LabeledList.tsx]: Unnecessary preceding whitespace here too

**Review by KubeRoot — CHANGES_REQUESTED** [review]

**Comment by KubeRoot** [comment]
Additionally, seems to add (presumably unused) sprites for tiered canisters, which we do not have: 
<details>
<summary>Screenshot</summary>

image

</details>

I don't know what `can-0` through `3` refer to here, might be unrelated, but `tier1` through `3` are almost certainly for tiered canisters.

**Comment by BrumalLake** [comment]
can-0-3 show the pressure inside the canister and should be kept

**Comment by Archanial** [comment]
Nothing broke during testing.

**Pull request merged**